feat: add @reboot scheduler support

foecmke opened the pull request.

foecmke committed:
feat: add @reboot scheduler support

gocronx commented:
@foecmke  谢谢提pr，但是需要麻烦优化下先:
1、并发安全，可以把 ran 改为 uint32 并用 sync/atomic 在 Next 中原子设置，避免竞态。
2、常量：把 3s 提取为 RebootDelay 常量
3、增加测试用例，增加*_test.go文件进行测试用例测试

foecmke committed:
优化：`@reboot` 调度器运行逻辑

- 将 ran 字段从 bool 类型改为 uint32 类型
- 使用 atomic.CompareAndSwapUint32 替代简单的布尔标志检查
- 引入 RebootDelay 常量替代硬编码的 3 秒延迟
- 添加 reboot_test.go 测试用例

foecmke commented:
还有什么需要完善的地方吗，感觉这项功能相当实用

gocronx commented:
已合并，欢迎提pr
